internal/cmd/dev/rslrecordat/rslrecordat.go

```diff
@@ -47,7 +47,18 @@ func (o *options) Run(_ *cobra.Command, args []string) error {
 		return err
 	}
 
-	return repo.RecordRSLEntryForReferenceAtTarget(args[0], o.targetID, signingKeyBytes)
+	if err := repo.RecordRSLEntryForReferenceAtTarget(args[0], o.targetID, signingKeyBytes); err != nil {
+		return err
+	}
+
+	if err := repo.VerifyRef(cmd.Context(), args[0], false); err != nil {
+		if err := repo.InvalidateLatestRSLEntry(); err != nil {
+			return err
+		}
+		return fmt.Errorf("verifcation of reference to be record failed, creation of RSL entry is not possible, err = %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("policy verification failed for ...: %w", err)
```

The `...` could possibly include the ref and the targetID at which verification failed.

Reply:
Fixed